hubstorage/client.py

```diff
@@ -28,10 +28,8 @@ def _hc_retry_on_exception(err):
         logger.warning("Server failed with %d status code, retrying (maybe)" % (err.response.status_code,))
         return True
 
-    # TODO: python3 compatibility: BadStatusLine error are wrapped differently
-    if (isinstance(err, ConnectionError) and err.args[0] == 'Connection aborted.' and
-            isinstance(err.args[1], BadStatusLine) and err.args[1][0] == repr('')):
-        logger.warning("Protocol failed with BadStatusLine, retrying (maybe)")
+    if isinstance(err, ConnectionError):
+        logger.warning("Request encountered a connection error: %s, retrying (maybe)" % (err,))
```

Comment:
Please use `%r` to format the error to retain error type information, as str only returns the message.

Also, logging module can format error messages, like `logger.warning("... %r ...", err)`, and postpone the formatting until it's actually needed. I see that has been done above, so might fix both cases while you're at it.